sdk/digitaltwins/Azure.DigitalTwins.Core/src/Models/DigitalTwinMetadataJsonConverter.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Azure.DigitalTwins.Core
+{
+#pragma warning disable AZC0014 // Avoid using banned types in public API
+
+    /// <summary>
+    /// JSON converter to make it easier to deserialize a <see cref="BasicDigitalTwin"/>.
+    /// </summary>
+    public class DigitalTwinMetadataJsonConverter : JsonConverter<DigitalTwinMetadata>
+    {
+        /// <inheritdoc/>
+        public override DigitalTwinMetadata Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
```

Comment:
I think this will work for the golden path, i'm wondering if we need to harden it against invalid/unexpected input.

Reply:
I'm worried about having enough time to put this in shape by 10/30. If this was the only thing we had left, I wouldn't be so worried, but that isn't the case.

Reply:
Understood. What are our options? Can we use the JsonExtensionData and then follow up in a subsequent minor version with one which is strongly typed? This feels like it would be a breaking change, would that be allowed?

Reply:
Changing the type would make it a breaking change. We probably couldn't change this specific type. We could introduce a new version of BasicDigitalTwin, which is kinda messy.

Reply:
Meant to say, no, breaking changes are not allowed.

Reply:
IOW, we're better off fixing an error condition later, than not shipping it now. :(

Reply:
> i'm wondering if we need to harden it against invalid/unexpected input.

What type of hardening were you thinking of? The JsonSerializer will handle "incomplete" data for you (at least for the sub-JSON relevant to this converter), along with invalid JSON input.

Do you want to validate that all the properties exist and are set?